libraries/lib-music-information-retrieval/MirDsp.cpp

```diff
@@ -74,13 +75,29 @@ std::vector<float> GetNormalizedCircularAutocorr(std::vector<float> x)
       return x;
    const auto N = x.size();
    assert(IsPowOfTwo(N));
-   PowerSpectrum(N, x.data(), x.data());
-   // We need the entire power spectrum for the auto-correlation, not only the
-   // left-hand side.
-   std::copy(x.begin() + 1, x.begin() + N / 2 - 1, x.rbegin());
```

Comment:
I notice now that the old version had an off-by-one mistake that escaped earlier review!  The bin one before Nyquist was not correctly mirrored.

Reply:
Indeed ...